### DISCUSSION TITLE
How to create signalStoreFeature similar to withComputed

### DISCUSSION THREAD

**Discussion by gmfun** [ngrx/platform#4162] [Q&A]
I want to create a signalStoreFeature similar to `withComputed` which works like bellow

```typescript
function withApiTableSource((store: WhatShouldThisBe): TableConfig) {
  return signalStoreFeature(
     withMethods(...),
     withComputed(...),
     withHooks()
  )
}

//usage
const store = signalStore(
  withState({a: 1}),
  withApiTableSource((store) => {
    // store.a should be Signal<number>
  })
)
```
`withApiTableSource` takes a function as an argument and the argument of this function is `store`, same store that is passed when I use `withComputed((store: THIS) => ...)` function.

I see in the sourcecode of `withComputed` it something like bellow

```typescript
export function withComputed<
  Input extends SignalStoreFeatureResult,
  Signals extends SignalsDictionary
>(
  signalsFactory: (
    store: Prettify<SignalStoreSlices<Input['state']> & Input['signals']>
  ) => Signals
)
```
But `SignalStoreFeatureResult`, `Prettify`, `SignalStoreSlices` etc are not exposed from the library.

**Comment by rainerhahnekamp** [comment]
That's not that easy. The store has pre-defined properties or "slots", where you can only add methods, computed, state or hooks. `withComputed`, `withMethods`, etc. have access to the internals, whereas `signalStoreFeature` is meant for us mortals ;)

Could you please post the full usage of `withApiTableSource`? Maybe I can help you if I see what is inside `withTableSource` and what the final properties of the store should look like.

**Comment by markostanimirovic** [comment]
This is possible, but not provided out of the box via `signalStoreFeature` function.

You need to 'manually' create a feature similar to base features, for example: https://github.com/ngrx/platform/blob/main/modules/signals/src/with-methods.ts
